dp/generate_prsn.py

```diff
@@ -19,23 +19,47 @@
 logger = logging.getLogger(__name__)
 
 
-def dry_run(filepaths):
+def dry_run(filepaths, output_to_file=False, workdir=None):
     '''Perform metadata checks on the input files'''
-    logger.info('Dry Run')
-    for filepath in filepaths.values():
-        logger.info('')
-        logger.info('File: {}'.format(filepath))
-        try:
-            dataset = CFDataset(filepath)
-        except Exception as e:
-            logger.exception('{}: {}'.format(e.__class__.__name__, e))
-
-        for attr in 'project model institute experiment ensemble_member'.split():
+    if output_to_file: # Used for wps process in thunderbird
+        output = ['Dry Run']
+        for filepath in filepaths.values():
+            output.append('File: {}'.format(filepath))
+            try:
+                dataset = CFDataset(filepath)
+            except Exception as e:
+                output.append('{}: {}'.format(e.__class__.__name__, e))
+
+            for attr in 'project model institute experiment ensemble_member'.split():
+                try:
+                    output.append('{}: {}'.format(attr, getattr(dataset.metadata, attr)))
+                except Exception as e:
+                    output.append('{}: {}: {}'.format(attr, e.__class__.__name__, e))
+            output.append('dependent_varnames: {}'.format(dataset.dependent_varnames()))
+
+        filename = os.path.join(workdir, 'dry.txt')
+        with open(filename, 'w') as f:
+            for line in output:
+                f.write('{}\n'.format(line))
+
+        return filename
+
+    else:
+        logger.info('Dry Run')
+        for filepath in filepaths.values():
+            logger.info('')
+            logger.info('File: {}'.format(filepath))
             try:
-                logger.info('{}: {}'.format(attr, getattr(dataset.metadata, attr)))
+                dataset = CFDataset(filepath)
             except Exception as e:
-                logger.info('{}: {}: {}'.format(attr, e.__class__.__name__, e))
-        logger.info('dependent_varnames: {}'.format(dataset.dependent_varnames()))
+                logger.exception('{}: {}'.format(e.__class__.__name__, e))
+
+            for attr in 'project model institute experiment ensemble_member'.split():
+                try:
+                    logger.info('{}: {}'.format(attr, getattr(dataset.metadata, attr)))
+                except Exception as e:
+                    logger.info('{}: {}: {}'.format(attr, e.__class__.__name__, e))
+            logger.info('dependent_varnames: {}'.format(dataset.dependent_varnames()))
```

Comment:
While this works, I think we need to find a way to save on redundant code.  Both sides of the `if/else` run the exact same process with the exception of the way the logs are output.

What we might want to try in this case is defining an `outputer` that we can use in the same way for both cases. Here is an example of what I mean:
```python3
nrados@pcic-2024:~$ python3
Python 3.6.9 (default, Apr 18 2020, 01:56:04) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import logging
>>> import sys
>>> 
>>> logger = logging.getLogger()
>>> logger.setLevel(logging.DEBUG)
>>> 
>>> handler = logging.StreamHandler(sys.stdout)
>>> handler.setLevel(logging.DEBUG)
>>> formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
>>> handler.setFormatter(formatter)
>>> logger.addHandler(handler)
>>> 
>>> output_items = []
>>> 
>>> outputer = logger.info
>>> outputer('Foo')
2020-06-04 13:40:22,993 - root - INFO - Foo
>>> outputer = output_items.append
>>> outputer('Bar')
>>> output_items
['Bar']
```
You can see in both instances we passed a string into the method the same way but it produced a different result. With this `outputer` we would only have one set of dry run code.